.github/workflows/main.yml

```diff
@@ -1,105 +1,59 @@
-  #  branches: [ master ]
+on:
+  # Runs on every push. If a push contains multiple commits, it will be ran on the latest one.
+  push:
 
-  # Allows you to run this workflow manually from the Actions tab
-  #workflow_dispatch:
+env:
+  COMPILE_JOBS: 2
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  build-release:
-    # The type of runner that the job will run on
+  build:
+    name: Build (${{ matrix.build_type }}-deal.ii:${{ matrix.dealii_version }})
     runs-on: ubuntu-latest
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
-    steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
-
+    strategy:
+      fail-fast: false
+      matrix:
+        build_type: ["Release", "Debug"]
+        dealii_version: ["master", "v9.3.0"]
+    
+    # Run steps in container of dealii's master branch
+    container:
+      image: dealii/dealii:${{ matrix.dealii_version }}-focal
 
-      # Runs a single command using the runners shell
-      - name: sanity pull-request
-        if: github.event_name != 'pull_request'
-        run: |
-          echo "Github actions is sane, running for branch : ${GITHUB_REF#refs/heads/}"
-
-
-
-      # Runs a set of commands using the runners shell
-      - name: compile and test pull-request
-        if: github.event_name != 'pull_request'
-        run: |
-          echo "pulling docker image for branch"
-          docker pull dealii/dealii:master-focal
-          echo "Docker image is good, ready for action!"
-          docker run dealii/dealii:master-focal /bin/sh -c "git clone https://github.com/lethe-cfd/lethe && cd lethe && git checkout ${GITHUB_REF#refs/heads/} && mkdir build && cd build && cmake ../ -DCMAKE_BUILD_TYPE=Release && make -j2 && ctest";
-          
-      - name: sanity master
-        if: github.event_name == 'pull_request'
+    steps:
+      - name: Setup
         run: |
-          echo "Github actions is sane, running for master"
+          # Github action require containers to run as root. We can't do this without breaking the mpi tests.
+          # To allow the checkout action to operate, we preemptively give access to the workspace directory manually. 
+          sudo chown -R $USER:$USER $GITHUB_WORKSPACE
 
-      - name: compile and test master
-        if: github.event_name == 'pull_request'
-        run: |
-          echo "pulling docker image for master"
-          docker pull dealii/dealii:master-focal
-          echo "Docker image is good, ready for action!"
-          docker run dealii/dealii:master-focal /bin/sh -c "git clone https://github.com/lethe-cfd/lethe && cd lethe && mkdir build && cd build && cmake ../ -DCMAKE_BUILD_TYPE=Release && make -j2 && ctest";
+          # Since dealii image doesn't include Node.js, we'll install it
+          sudo apt-get update
+          sudo apt-get install -y --no-install-recommends nodejs
```

Comment:
Why the need for nodejs?

Reply:
The checkout action is written in nodejs. Without it we cannot checkout the code the way Github recommends it.